source/FingerBoard-Core/FBParallelNode.class.st

```diff
@@ -13,3 +13,10 @@ FBParallelNode >> activationClass [
 FBParallelNode >> defaultBehaviorClass [
 	^ FBParallelNodeBehavior 
 ]
+
+{ #category : #accessing }
+FBParallelNode >> outputFor: anActivation [
+	
+	"unlike other nodes, the parallel node output return the current input value being processed"
+	^ anActivation value at: anActivation instance executor index
```

Comment:
It is unlikely we ask for the output of an activation while it is in processing. So it is more likely the instance answers a different executor. And if the executor has been finished we would rather use the first element then the last